### PR TITLE
Updated the github workflow to generate the JSON schemas and Definitions for OSM components

### LINKED ISSUE #93: [CI] Patterns: Generate JSON Schema and Definitions for OSM components
**Enhancement Description**

To generate the workflow of getting the manifest from the OSM using the [kubeopenapi-jsonschema](https://github.com/layer5io/kubeopenapi-jsonschema) to read and generate json schemas from those manifests and then also generating workload definitions from those schemas

A sample workflow is [here](https://github.com/DelusionalOptimist/meshery-istio/blob/gen-oam/.github/workflows/update-oam-defs.yml)

---

### PR DESCRIPTION
**Pull request by sayantan1413**
Signed-off-by: sayantan1413 <sayantanbose2001@gmail.com>

**Description**

Updated the github workflow to generate JSON Schema and Definitions for OSM components

This PR fixes #93 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@utkarsh-pro review when you have a few moments, if you would.

**Comment by DelusionalOptimist** [comment]
Diff for the commit containing generated components: https://github.com/sayantan1413/meshery-osm/commit/7ede264abde811e2f488541c400062c87467137c#

**Comment by sayantan1413** [comment]
Hey @DelusionalOptimist is there anything odd ?

**Review by DelusionalOptimist — COMMENTED** [review]
@utkarsh-pro @leecalcote 
https://github.com/sayantan1413/meshery-osm/blob/7ede264abde811e2f488541c400062c87467137c/templates/oam/workloads-v0.9.1/egressosm.meshery.layer5io.schema.json#L104
Is it fine to use OSM? SMP protos call it `OPEN_SERVICE_MESH`

**Comment by DelusionalOptimist** [comment]
> Hey @DelusionalOptimist is there anything odd ?

Looking good @sayantan1413, just some comments for other reviewers :smile:

**Comment by leecalcote** [comment]
> @utkarsh-pro @leecalcote
> https://github.com/sayantan1413/meshery-osm/blob/7ede264abde811e2f488541c400062c87467137c/templates/oam/workloads-v0.9.1/egressosm.meshery.layer5io.schema.json#L104
> Is it fine to use OSM? SMP protos call it `OPEN_SERVICE_MESH`

Unless using the long form of the service mesh name messes things up, yes, ideally, we're using what is defined in the spec.

Why is there no separator between the name of the object and the name of the service mesh (e.g. egressosm --> egress.osm)?

**Comment by DelusionalOptimist** [comment]
> Why is there no separator between the name of the object and the name of the service mesh (e.g. egressosm --> egress.osm)?

Hmm :thinking:. Not sure about this.
We don't have separators in any of the old, self-added Istio components
https://github.com/meshery/meshery-istio/blob/4376fa99dc638285b48f4d1392e7a8091028865b/templates/oam/workloads/envoyfilter_definition.json#L5
So we tried to do the same here too.
But I think we should have separators. What do you say @utkarsh-pro

**Comment by sayantan1413** [comment]
Yup I also think the seperator would be better here.🤔

**Comment by leecalcote** [comment]
I assume that lack of separation is not for any other reason than @utkarsh-pro didn't get to it, yet. There's much merit to having a separator so that proper classification and display of these operations can ensue.

**Comment by utkarsh-pro** [comment]
Yeah, we had to place separators in the names of the components. A related issue regarding the same was/is being tracked here: https://github.com/meshery/meshery-istio/issues/229

// @DelusionalOptimist @sayantan1413 @leecalcote

**Comment by leecalcote** [comment]
Who is taking this issue forward?

**Comment by sayantan1413** [comment]
Hey @leecalcote I would like to try out a bit to put a separator.

**Comment by DelusionalOptimist** [comment]
Thanks @sayantan1413 :slightly_smiling_face:

**Comment by leecalcote** [comment]
How'd you make out, @sayantan1413?

File: .github/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+CURRENT_OSM_VERSION=v0.9.0
```
Comment (leecalcote) [inline .github/.env:1]: Can we kill this? Each individual component should have a version number either in the component's json or in the folder/filename (ideally in the json as this needs to be included in the capabilities registry in meshery server). @Shreyas220 @DelusionalOptimist @sayantan1413 @utkarsh-pro @Revolyssup

Comment (DelusionalOptimist) [reply .github/.env:1]: Yes @leecalcote we can kill this. 
This was the hastiest idea I could come up to make a workflow trigger consistently with a service mesh's new releases.
Just to clarify, we are storing service mesh version in component's schema even right now and probably the appropriate approach will be reading them from the schema itself for triggering component generation.

**Pull request closed without merging**